apps/laboratory/src/components/Ethers/EthersGetCallsStatusTest.tsx

```diff
@@ -0,0 +1,105 @@
+import { Button, Stack, Text, Input } from '@chakra-ui/react'
+import { useState } from 'react'
+import { useWeb3ModalAccount, useWeb3ModalProvider } from '@web3modal/ethers/react'
+import { EthereumProvider } from '@walletconnect/ethereum-provider'
+import { useChakraToast } from '../Toast'
+import { BrowserProvider } from 'ethers'
+import { type GetCallsStatusParams } from '../../types/EIP5792'
+import { EIP_5792_RPC_METHODS, getAtomicBatchSupportedChainInfo } from '../../utils/EIP5792Utils'
+
+export function EthersGetCallsStatusTest() {
+  const toast = useChakraToast()
+  const { address, chainId } = useWeb3ModalAccount()
+  const { walletProvider } = useWeb3ModalProvider()
+  const [loading, setLoading] = useState(false)
+  const ethereumProvider =
+    walletProvider && (walletProvider as Awaited<ReturnType<(typeof EthereumProvider)['init']>>)
```

Comment:
isn't this setting a boolean flag on `ethereumProvider` ? Or is it resolving to the typed version of `walletProvider`?

If that's the case I believe it'd be best to do the check + cast whenever this `ethereumProvider` is used directly.

Reply:
Done